sklearn/gaussian_process/_gpc.py

```diff
@@ -777,12 +850,28 @@ def predict_proba(self, X):
                 "one_vs_rest mode instead."
             )
 
+        can_return_f_std = isinstance(
+            self.base_estimator_, _BinaryGaussianProcessClassifierLaplace
+        )
+        if not can_return_f_std and return_std_of_f:
+            error_msg = (
+                "Returning the standard deviation of the "
+                "latent function f is only supported for GPCs "
+                "that use the Laplace Approximation."
+            )
+            raise ValueError(error_msg)
+
         if self.kernel is None or self.kernel.requires_vector_input:
             X = self._validate_data(X, ensure_2d=True, dtype="numeric", reset=False)
         else:
             X = self._validate_data(X, ensure_2d=False, dtype=None, reset=False)
 
```

Comment:
I would put the check whether `std_f` can be returned or not here, after the kernel tests. Than the check can be similar to the one done around lines 773:

`if self.n_classes_ > 2:`
 `    if return_std_of_f:`
`         raise ValueError(`
                `         "Returning the standard deviation of the "`
               `          "latent function f is only supported for GPCs "`
                `         "that use the Laplace Approximation."`
            `         )`
`    else:`
`        return self.base_estimator_.predict_proba(X)`
`else:`
`    return self.base_estimator_.predict_proba(`
`                X, return_std_of_f=return_std_of_f`
`            )`

Reply:
Agreed! Will implement these changes.